7_react/react1/week3/todo/src/todoList/TodoList.jsx

```diff
@@ -0,0 +1,121 @@
+import TodoItem from "../todoItem/TodoItem";
+import { useEffect, useState } from "react";
+import Border from "../border/Border";
+
+function AppTodo() {
+  const [todolist, setTodoList] = useState([]);
+ 
+  useEffect(() => {
+    fetch(
+      "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+    )
+      .then((response) => response.json())
+      .then((data) => setTodoList(data))
+      .catch((error) => console.error("Error fetching todos:", error));
+  }, []);
+
+  const addTodo = (description, deadline) => {
+    if (description && deadline) {
+      const newTodo = {
+        id: new Date(),
+        description: description,
+        deadline: deadline,
+        status: false,
+      };
+      setTodoList((prev) => [...prev, newTodo]);
+    } else {
+      alert('Please fill all the lines!')
+    }
+  };
+
+  
+  const onDeleteClick = (id) => {
+    setTodoList(todolist.filter((todo) => todo.id !== id));
+  };
+  
+  const toggleStatus = (id, status) => {
+    const update = todolist.map((todo) =>
+    todo.id === id ? { ...todo, status: !status } : todo
+    );
+    setTodoList(update);
+    console.log(todolist);
+  };
+  
+  const TodoList = (props) => {
+    return (
+      <div>
+        <ul>
+            {props.todos.length > 0 ? (
+              props.todos.map((todo) => {
+                return (
+                 
+                    <TodoItem
+                      key={todo.id}
+                      description={todo.description}
+                      deadline={todo.deadline}
+                      status={todo.status}
+                      onDeleteClick={onDeleteClick}
+                      toggleStatus={toggleStatus}
+                    />
+                  
+                );
+              })
+            ) : (
+              <p>No items...</p>
+            )}
+        </ul>
+      </div>
+    );
+  };
+  
+  
+  const AddTodoForm = () => {
```

Comment:
You have already another file with this component, which is really good. I recommend that you remove it from this file and import it from `AddTodo.jsx`